mobile/mobile_46.asm

```diff
@@ -6819,13 +6821,13 @@ Mobile46_RunJumptable:
 	jumptable .Jumptable, wJumptableIndex
 
 .Jumptable:
-	dw Function11b483
-	dw Function11b570
-	dw Function11b5c0
-	dw Function11b5e0
+	dw PrepareMonForUpload
+	dw Function11b570 ; ???
+	dw Function11b5c0 ; Remove from party
+	dw Function11b5e0 ; Set script value to 0, indicates success
```

Comment:
Whatever this jumptable's for should probably have a `Prefix_` for all of these, they're otherwise pretty generic routine names.
```suggestion
	dw RemoveMonFromParty
	dw Success
```

Reply:
`TradeCorner_`?

Reply:
I believe this is the mobile trade feature in the pokecom center, but its another case of something i documented 2 years ago and don't really remember now

If the comments are an issue i'd prefer to omit them entirely over giving symbols potentially misleading names

Reply:
```assembly
Function11b444:
; special
	call Mobile46_InitJumptable
	call Mobile46_RunJumptable
	ret
```
This is the special that calls the jumptable in question. It appears that It was supposed to have been called from the GoldenrodPokecenter1FTradeCornerAttendant. It performs the actual trade. Returns $00 on success. See the following:
https://github.com/Sudel-Matze/pokecrystal/blob/b08ed8346723f57e56489e34bf558d07c9b20f70/maps/GoldenrodPokecenter1F.asm#L101-L110

`db $0F` is the special command.. And `db $6e` corresponds to `Function11b444Special`.

So.. as far as a name/prefix .. we should do some combination of Mobile, TradeCorner, and Trade.

@Rangi42 Suggestions on a prefix standard going forward?

Reply:
I have a complete version of the trade corner script
https://github.com/ariscop/pokecrystal/blob/pokecom/maps/GoldenrodPokecomCenter.asm#L44
fartext isnt in jp crystal and a lot of script opcodes are shifted as a result

Reply:
Ahhhh okay this makes more sense....

`Function11b444:` is responsible for the TradeCorner HoldMon special script.

Reply:
If this jumptable is for multiple things (mobile trading, the trade corner, maybe the news machine too, mobile battling, etc) it's fine if each has its own prefix. Mostly I want to avoid giving unused mobile-only code names that sound generic and useful in regular code (e.g. "``RemoveMonFromParty`").

Reply:
Lets do this:

For `Function11b444:`
```asm
Mobile_TradeCornerHoldMon:
; special
	call Mobile46_InitJumptable ; This is used elsewhere.
	call Mobile_TradeCornerHoldMon_RunJumptable
	ret
```

For the `.Jumptable:` (We should turn these into local routines.)
```
.Jumptable:
	dw .PrepareMonForUpload
	dw .Function11b570 ; ???
	dw .RemoveMonFromParty ; Remove from party
	dw .Success
	dw .Noop ; unused
```

Reply:
As long as those functions aren't too complex with their own local labels, then yeah, sure.

Reply:
WIP local label renames for .PrepareMonForUpload
```asm
.PrepareMonForUpload:
	call .InitRAM
	ld hl, wPlayerName
	ld a, NAME_LENGTH_JAPANESE - 1
.get_char
	push af
	ld a, [hli]
	ld [bc], a
	inc bc
	pop af
	dec a
	and a
	jr nz, .get_char

	ld de, PARTYMON_STRUCT_LENGTH
	ld hl, wPartyMon1Species
	ld a, [wcd82]
	dec a
	push af
.get_next_party_mon
	and a
	jr z, .got_selected_mon
	add hl, de
	dec a
	jr .get_next_party_mon

.got_selected_mon
	push bc
	ld a, PARTYMON_STRUCT_LENGTH

.copy_mon_byte
	; Copies Party Mon into [bc]
	push af
	ld a, [hli]
	ld [bc], a
	inc bc
	pop af
	dec a
	and a
	jr nz, .copy_mon_byte
...
```

Reply:
would discourage local labels here tbh.

Reply:
> As long as those functions aren't too complex with their own local labels, then yeah, sure.

They don't seem to be; I think we can easily keep their own local labels different. In this particular case.
This is mainly to avoid excessively long label names. However, if you are fine with the long label names we don't need to do locals.

Reply:
When it comes to jumptables I'm generally more concerned about having to carefully pick or prefix locals than having to prefix each jumptable entry. That is unless the branches are trivial and/or they jump or fall through into one another. This doesn't fit that description, and it's in a file with dozens of jumptables referencing functions across the file anyway, might as well make them match up.

Reply:
So... then `Mobile_TradeCornerHoldMonPrepareMonForUpload` is fine?

Reply:
I'd like to reserve `Mobile_` for the mobile SDK and the mobile_45.asm email apis. TradeCornerHoldMon_PrepareForUpload is fine.